File: eDental-flutter-main/lib/models/rating.dart
import 'package:dart_json_mapper/dart_json_mapper.dart';
import 'package:edental/helpers/dateJsonFormatter.dart';

@jsonSerializable
class Rating {
  int? id;
  int rate;
  String comment;
  @JsonProperty(converter: CustomDateTimeConverter())
  DateTime date;
  int userId;
  int dentistId;
  @JsonProperty(ignoreIfNull: true)
  String? dentistFullName;
  @JsonProperty(ignoreIfNull: true)
  String? clientFullName;

  Rating(
    this.rate,
    this.comment,
    this.date,
    this.userId,
    this.dentistId, {
    this.id,
    this.dentistFullName,
    this.clientFullName,
  });
}

File: edentalbackend-main/DentalOffice.Entities/dbEntities/Rating.cs
using System.ComponentModel.DataAnnotations.Schema;

namespace DentalOffice.Entities.dbEntities
{
    public class Rating
    {
        public int Id { get; set; }
        public int Rate { get; set; }
        public string Comment { get; set; }
        public DateTime Date { get; set; }
        public int UserId { get; set; }
        public User User { get; set; }
        public int DentistId { get; set; }
        public Dentist Dentist { get; set; }

        [NotMapped]
        public string DentistFullName { get; set; }
        [NotMapped]
        public string ClientFullName { get; set; }
    }
}
